offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/OfferToPDFConverter.groovy

```diff
@@ -226,60 +221,64 @@ class OfferToPDFConverter implements OfferExporter {
-            final String formattedOverheadPrice = Currency.getFormatterWithoutSymbol().format(overheadPrice)
+            //Map Lists to the "DataGeneration", "DataAnalysis" and "Project and Data Management"
+            productItemsMap.dataGeneration = dataGenerationItems
+            productItemsMap.dataAnalysis = dataAnalysisItems
+            productItemsMap.dataManagement= dataManagementItems
 
-            htmlContent.getElementById("${overheadStatus}-value-total").text(formattedTotalPrice)
-            htmlContent.getElementById("${overheadStatus}-value-net").text(formattedItemsWithOverheadNetPrice)
-            htmlContent.getElementById("${overheadStatus}-value-overhead").text(formattedOverheadPrice)
+            return productItemsMap
         }
 
-    void generateProductTable(String overheadStatus, List<ProductItem> productItems, int maxTableItems){
+    void generateProductTable(Map productItemsMap, int maxTableItems) {
         // Create the items in html in the overview table
-        if (!productItems.isEmpty()) {
-        // set initial product item position in table
-        def itemPos = 1
-        // max number of table items per page
-        def elementId = "product-items" + "-" + tableCount
-            //Append Table Title
-            htmlContent.getElementById(elementId).append(ItemPrintout.tableTitle(overheadStatus))
-            productItems.each { productItem ->
-                //start (next) table and add Product to it
-                if (tableItemsCount >= maxTableItems)
-                {
-                    ++tableCount
-                    elementId = "product-items" + "-" + tableCount
-                    htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
-                    tableItemsCount = 1
+        productItemsMap.each {key, value ->
+            //Check if there are ProductItems stored in map entry
+            if (value) {
+                // set initial product item position in table
+                def itemPos = 1
+                // max number of table items per page
+                def elementId = "product-items" + "-" + tableCount
+                //Append Table Title
+                htmlContent.getElementById(elementId).append(ItemPrintout.tableTitle(key.toString()))
+                value.each {productItem ->
+                    //start (next) table and add Product to it
+                    if (tableItemsCount >= maxTableItems) {
+                        ++tableCount
+                        elementId = "product-items" + "-" + tableCount
+                        htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
+                        tableItemsCount = 1
+                    }
+                    //add product to current table
+                    htmlContent.getElementById(elementId).append(ItemPrintout.itemInHTML(itemPos++, productItem as ProductItem))
```

Comment:
Are you aware of this interface method of Collections? 

http://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Object.html#eachWithIndex(groovy.lang.Closure)

You can simplify this code block significantly and make it more readable.

And: upgrading variable inline, like ``itemPos++`` is hard to debug and not very expressive,

Reply:
Thanks for the suggestion, i will use it in the future and added it 👍